pkg/api/management/api/alert/notifier_action.go

```diff
@@ -0,0 +1,376 @@
+			s := notifier.Spec.SMTPConfig
+			return testEmail(s.Host, s.Password, s.Username, int(s.Port), s.TLS, msg, s.DefaultRecipient)
+		}
+
+		if notifier.Spec.PagerdutyConfig.ServiceKey != "" {
+			return testPagerduty(notifier.Spec.PagerdutyConfig.ServiceKey, msg)
+		}
+
+		if notifier.Spec.WebhookConfig.URL != "" {
+			return testWebhook(notifier.Spec.WebhookConfig.URL, msg)
+		}
+
+	} else {
+
+		slackConfigInterface, exist := actionInput["slackConfig"]
+		if exist {
+			slackConfig, ok := slackConfigInterface.(map[string]interface{})
+			if ok {
+				url := slackConfig["url"].(string)
+				return testSlack(url, msg)
+			}
+		}
+
+		smtpConfigInterface, exist := actionInput["smtpConfig"]
+		if exist {
+			smtpConfig, ok := smtpConfigInterface.(map[string]interface{})
+			if ok {
+				host := smtpConfig["host"].(string)
+				port, _ := smtpConfig["port"].(json.Number).Int64()
+				password := smtpConfig["password"].(string)
+				username := smtpConfig["username"].(string)
+				receiver := smtpConfig["defaultRecipient"].(string)
+				tls := smtpConfig["tls"].(bool)
+				return testEmail(host, password, username, int(port), tls, msg, receiver)
+			}
+		}
+
+		webhookConfigInterface, exist := actionInput["webhookConfig"]
+		if exist {
+			webhookConfig, ok := webhookConfigInterface.(map[string]interface{})
+			if ok {
+				url := webhookConfig["url"].(string)
+				return testWebhook(url, msg)
+			}
+		}
+
+		pagerdutyConfigInterface, exist := actionInput["pagerdutyConfig"]
+		if exist {
+			pagerdutyConfig, ok := pagerdutyConfigInterface.(map[string]interface{})
+			if ok {
+				key := pagerdutyConfig["serviceKey"].(string)
+				return testPagerduty(key, msg)
+			}
+		}
+
+	}
+
+	return nil
+}
+
+type pagerDutyMessage struct {
+	RoutingKey  string `json:"routing_key,omitempty"`
+	ServiceKey  string `json:"service_key,omitempty"`
+	DedupKey    string `json:"dedup_key,omitempty"`
+	IncidentKey string `json:"incident_key,omitempty"`
+	EventType   string `json:"event_type,omitempty"`
+	Description string `json:"description,omitempty"`
+}
+
+func hashKey(s string) string {
+	h := sha256.New()
+	h.Write([]byte(s))
+	return fmt.Sprintf("%x", h.Sum(nil))
+}
+
+func testPagerduty(key, msg string) error {
+	if msg == "" {
+		msg = "test pagerduty service key"
+	}
+
+	pd := &pagerDutyMessage{
+		ServiceKey:  key,
+		EventType:   "trigger",
+		IncidentKey: hashKey("key"),
+		Description: msg,
+	}
+
+	url := "https://events.pagerduty.com/generic/2010-04-15/create_event.json"
+
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(pd); err != nil {
+		return err
+	}
+	resp, err := http.Post(url, "application/json", &buf)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("http status code is not 200")
+	}
+
+	return nil
+}
+
+func testWebhook(url, msg string) error {
+	if msg == "" {
+		msg = "test webhook"
+	}
+	alertList := model.Alerts{}
+	a := &model.Alert{}
```

Comment:
Nit: see if initializing can be done in 1 line:

```
a := &model.Alert{
a.Labels: map[model.LabelName]model.LabelValue{
"test_msg": model.LabelValue(msg)
}
}
```

Reply:
yep, I changed it to:
```
alertList := model.Alerts{
		&model.Alert{
			Labels: map[model.LabelName]model.LabelValue{
				model.LabelName("test_msg"): model.LabelValue(msg),
			},
		},
	}
```